src/main/java/com/google/api/generator/gapic/composer/samplecode/SampleComposer.java

```diff
@@ -0,0 +1,189 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer.samplecode;
+
+import com.google.api.generator.engine.ast.AssignmentExpr;
+import com.google.api.generator.engine.ast.ClassDefinition;
+import com.google.api.generator.engine.ast.Expr;
+import com.google.api.generator.engine.ast.ExprStatement;
+import com.google.api.generator.engine.ast.MethodDefinition;
+import com.google.api.generator.engine.ast.MethodInvocationExpr;
+import com.google.api.generator.engine.ast.ScopeNode;
+import com.google.api.generator.engine.ast.Statement;
+import com.google.api.generator.engine.ast.TypeNode;
+import com.google.api.generator.engine.ast.Variable;
+import com.google.api.generator.engine.ast.VariableExpr;
+import com.google.api.generator.gapic.composer.comment.CommentComposer;
+import com.google.api.generator.gapic.model.RegionTag;
+import com.google.api.generator.gapic.model.Sample;
+import com.google.api.generator.gapic.utils.JavaStyle;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SampleComposer {
+  public static String createInlineSample(List<Statement> sampleBody) {
+    List<Statement> statementsWithComment = new ArrayList<>();
+    statementsWithComment.addAll(CommentComposer.AUTO_GENERATED_SAMPLE_COMMENT);
+    statementsWithComment.addAll(sampleBody);
+    return SampleCodeWriter.write(statementsWithComment);
+  }
+
+  //  "Executable" meaning it includes the necessary code to execute java code,
+  //  still may require additional configuration to actually execute generated sample code
+  public static String createExecutableSample(Sample sample, String pakkage) {
+    Preconditions.checkState(!sample.fileHeader().isEmpty(), "File header should not be empty");
+    String sampleHeader = SampleCodeWriter.write(sample.fileHeader());
+    String sampleClass =
+        SampleCodeWriter.write(
+            composeExecutableSample(
+                pakkage,
+                sample.name(),
+                sample.variableAssignments(),
+                bodyWithComment(CommentComposer.AUTO_GENERATED_SAMPLE_COMMENT, sample.body())));
+    sampleClass = includeRegionTags(sampleClass, sample.regionTag());
+    return String.format("%s\n%s", sampleHeader, sampleClass);
+  }
+
+  private static List<Statement> bodyWithComment(
+      List<Statement> autoGeneratedComment, List<Statement> sampleBody) {
+    List<Statement> bodyWithComment = new ArrayList<>();
+    bodyWithComment.addAll(autoGeneratedComment);
+    bodyWithComment.addAll(sampleBody);
+    return bodyWithComment;
+  }
+
+  @VisibleForTesting
+  protected static String includeRegionTags(String sampleClass, RegionTag regionTag) {
+    Preconditions.checkState(
```

Comment:
apiShortName does not get set while generating samples because that context is not passed in while generating classes. I could pass that context all the places I'm generating classes or I can just set the apiShortName after I've generated classes in the [Composer](https://github.com/googleapis/gapic-generator-java/pull/935/files#diff-a1633e68314b79aea28ed9f01fc083722777c0c5f02b1ba8a85fd6f6c5355be6R191). It probably makes more sense to just not have this be an attribute of regionTag or not have the concept of regionTag and just pass this needed argument in when writing an executable sample, but for readability I kind of liked it tied to RegionTag since this attribute is specifically required to write a regionTag.